presto-geospatial/src/test/java/com/facebook/presto/plugin/geospatial/TestGeoFunctions.java

```diff
@@ -1019,6 +1020,61 @@ public void testSTLineString()
         assertFunction("ST_LineString(array[ST_Point(1,2), ST_GeometryFromText('POINT EMPTY'), ST_Point(3,4), ST_GeometryFromText('POINT EMPTY')])", GEOMETRY, "LINESTRING (1 2, 3 4)");
     }
 
+    @Test
+    public void testMultiPoint()
+    {
+        // General case, 2+ points
+        assertMultiPoint("MULTIPOINT ((1 2), (3 4))", "POINT (1 2)", "POINT (3 4)");
+        assertMultiPoint("MULTIPOINT ((1 2), (3 4), (5 6))", "POINT (1 2)", "POINT (3 4)", "POINT (5 6)");
+        assertMultiPoint("MULTIPOINT ((1 2), (3 4), (5 6), (7 8))", "POINT (1 2)", "POINT (3 4)", "POINT (5 6)", "POINT (7 8)");
+
+        // Duplicate points work
+        assertMultiPoint("MULTIPOINT ((1 2), (1 2))", "POINT (1 2)", "POINT (1 2)");
+        assertMultiPoint("MULTIPOINT ((1 2), (3 4), (1 2))", "POINT (1 2)", "POINT (3 4)", "POINT (1 2)");
+
+        // Single point
+        assertMultiPoint("MULTIPOINT ((1 2))", "POINT (1 2)");
+
+        // Empty array
+        assertFunction("ST_MultiPoint(array[])", GEOMETRY, null);
+
+        // Only points can be passed
+        assertInvalidMultiPoint("geometry is not a point: LineString at index 2", "POINT (7 8)", "LINESTRING (1 2, 3 4)");
+
+        // Null point raises exception
+        assertInvalidMultiPoint("null at index 1", "null");
+        assertInvalidMultiPoint("null at index 3", "POINT (1 2)", "POINT (1 2)", "null");
+        assertInvalidMultiPoint("null at index 2", "POINT (1 2)", "null", "POINT (3 4)");
+        assertInvalidMultiPoint("null at index 2", "POINT (1 2)", "null", "POINT (3 4)", "null");
+
+        // Empty point raises exception
+        assertInvalidMultiPoint("empty point at index 1", "POINT EMPTY");
+        assertInvalidMultiPoint("empty point at index 2", "POINT (1 2)", "POINT EMPTY");
+    }
+
+    private void assertMultiPoint(String expectedWkt, String... pointWkts)
+    {
+        assertFunction(
+                format(
+                        "ST_MultiPoint(array[%s])",
+                        ImmutableList.copyOf(pointWkts).stream()
+                            .map(wkt -> wkt.equalsIgnoreCase("null") ? "null" : format("ST_GeometryFromText('%s')", wkt))
```

Comment:
Why not use real nulls for this instead of doing a string match?

Reply:
@martint `ImmutableList.copyOf` doesn't allow nulls. I could use `Arrays.asList` though. Would you think this better?

```
        // Null point raises exception
        assertInvalidFunction("ST_MultiPoint(array[null])", "Invalid input to ST_MultiPoint: null at index 1");
        assertInvalidMultiPoint("null at index 3", "POINT (1 2)", "POINT (1 2)", null);
        assertInvalidMultiPoint("null at index 2", "POINT (1 2)", null, "POINT (3 4)");
        assertInvalidMultiPoint("null at index 2", "POINT (1 2)", null, "POINT (3 4)", null);

    private void assertInvalidMultiPoint(String errorMessage, String... pointWkts)
    {
        assertInvalidFunction(
                format(
                        "ST_MultiPoint(array[%s])",
                        Arrays.asList(pointWkts).stream()
                            .map(wkt -> wkt == null ? "null" : format("ST_GeometryFromText('%s')", wkt))
                            .collect(Collectors.joining(","))),
                INVALID_FUNCTION_ARGUMENT,
                format("Invalid input to ST_MultiPoint: %s", errorMessage));
    }
```

Reply:
Since you're not using the `List` returned from `Arrays.asList`/`ImmutableList.copyOf` for anything other than calling `stream()`, you can just do:

```
Arrays.stream(pointWkts)
    .map(...)
    ...
```

Reply:
@martint Thanks for the suggestion. Updated.